tests/indexes/test_api.py

```diff
@@ -21,62 +21,138 @@
+                    "job": {"id": "bar"},
+                    "reference": {"id": "bar"},
+                    "user": {"id": user.id},
+                    "sequence_otu_map": {"foo": "bar_otu"},
+                },
+                {
+                    "_id": "foo",
+                    "version": 0,
+                    "created_at": static_time.datetime,
+                    "manifest": {"foo": 2},
+                    "ready": False,
+                    "has_files": True,
+                    "job": {"id": "foo"},
+                    "reference": {"id": "foo"},
+                    "user": {"id": user.id},
+                    "sequence_otu_map": {"foo": "foo_otu"},
+                },
+            ]
+        )
 
-    await client.db.indexes.insert_many(
-        [
+        await client.db.history.insert_many(
+            [
+                {"_id": "0", "index": {"id": "bar"}, "otu": {"id": "baz"}},
+                {"_id": "1", "index": {"id": "foo"}, "otu": {"id": "baz"}},
+                {"_id": "2", "index": {"id": "bar"}, "otu": {"id": "bat"}},
+                {"_id": "3", "index": {"id": "bar"}, "otu": {"id": "baz"}},
+                {"_id": "4", "index": {"id": "bar"}, "otu": {"id": "bad"}},
+                {"_id": "5", "index": {"id": "foo"}, "otu": {"id": "boo"}},
+            ]
+        )
+
+        mocker.patch(
+            "virtool.indexes.db.get_unbuilt_stats",
+            make_mocked_coro(
+                {"total_otu_count": 123, "change_count": 12, "modified_otu_count": 3}
+            ),
+        )
+
+        resp = await client.get("/indexes")
+
+        assert resp.status == 200
+        assert await resp.json() == snapshot
+
+    async def test_ready(self, snapshot, fake2, spawn_client, static_time):
+        client = await spawn_client(authorize=True)
+
+        user = await fake2.users.create()
+
+        await client.db.indexes.insert_many(
+            [
+                {
+                    "_id": "bot",
+                    "change_count": 2,
+                    "modified_otu_count": 4,
+                    "version": 1,
+                    "created_at": static_time.datetime,
+                    "manifest": {"foo": 2},
+                    "ready": True,
+                    "has_files": True,
+                    "job": {"id": "bar"},
+                    "reference": {"id": "bar"},
+                    "user": {"id": user.id},
+                },
+                {
+                    "_id": "daz",
+                    "change_count": 3,
+                    "modified_otu_count": 5,
+                    "version": 0,
+                    "created_at": static_time.datetime,
+                    "manifest": {"foo": 2},
+                    "ready": True,
+                    "has_files": True,
+                    "job": {"id": "foo"},
+                    "reference": {"id": "foo"},
+                    "user": {"id": user.id},
+                },
+            ]
+        )
```

Comment:
Ineffective. Both fake indexes have ready set `true` so it doesn't actually test that non-ready indexes are filtered out of the result.

I would combine this with the above test and use parametrization to vary the value for `ready`.